tests/neg/i5954.scala

```diff
@@ -0,0 +1,10 @@
+abstract class MatcherFactory1 {
+  class AndNotWord
+}
+
+object MatcherFactory1 {
+  import scala.quoted._
+
+  def impl2(a: MatcherFactory1)(self: Expr[a.AndNotWord]) =
+  '{ ~self } // error: access to value a from wrong staging level
+}
```

Comment:
I'm wondering how we can use `a` properly in such context? We need to have some principles to deal with terms in types.

Reply:
Maybe we could encode with something like

```scala
def impl2[T: Type](a: MatcherFactory1)(self: Expr[T])(implicit ev: T =:= a.AndNotWord) = ...
```